op-service/sources/eth_client.go

```diff
@@ -537,3 +537,33 @@ func (s *EthClient) PendingNonceAt(ctx context.Context, account common.Address)
 	err := s.client.CallContext(ctx, &result, "eth_getTransactionCount", account, "pending")
 	return uint64(result), err
 }
+
+// BalanceAt returns the wei balance of the given account.
+func (s *EthClient) BalanceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (*big.Int, error) {
+	var result hexutil.Big
+	var err error
+	if blockNumber != nil {
+		err = s.client.CallContext(ctx, &result, "eth_getBalance", account, blockNumber)
+	} else {
+		err = s.client.CallContext(ctx, &result, "eth_getBalance", account, "latest")
+	}
+	return (*big.Int)(&result), err
+}
+
+// WaitTransaction waits transaction until the context is cancelled.
+func (s *EthClient) WaitTransaction(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
```

Comment:
I think this should go into the `wait` package in e2eutils as standalone function, rather than being attached to the `EthClient`

The `wait` package is meant to make a lot of these interactions with transaction/receipt/block-waiting easy.

Reply:
Fixed at https://github.com/ethereum-optimism/optimism/pull/14751/commits/7f634aca0777b824de669629bb76a11dafc88b55

but as we can see we e2e util heavily depends on ethclient.Client instead of sources.EthClient. In this case if we want complete removal of geth ethclient.Client from devnet-sdk, it seems to be requiring reimplementation of op-e2e utils is necessary.